subsys/net/l2/ethernet/ethernet.c

```diff
@@ -566,17 +559,18 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 		goto error;
 	}
 
-	if (IS_ENABLED(CONFIG_NET_STATISTICS_ETHERNET)) {
-		net_pkt_ref(pkt);
-	}
-
 	ret = api->send(net_if_get_device(iface), pkt);
-	if (!ret) {
-		ret = net_pkt_get_len(pkt);
```

Comment:
ethernet_send() needs to return the length sent on success

Reply:
OK, fixed.